synapseformation/schemas/long.schema.yml

```diff
@@ -0,0 +1,155 @@
+$base: "https://synapse.org#"
+
+$namespaces:
+  syn: "https://synapse.org#"
+
+$graph:
+
+- name: RootDoc
+  type: documentation
+  doc:
+    - |
+      # SynapseFormation Long Template
+
+      TBD
+      
+
+- name: ResourceRecord
+  doc: |
+    The abstract base type for a Synapse resource.
+  type: record
+  abstract: true
+  fields:
+  - name: name
+    doc: ""
+    type: string?
+  - name: id
+    doc: "Synapse ID"
+    type: string?
+
+- name: EntityRecord
+  doc: |
+    The abstract base type for a Synapse entity.
+  type: record
+  extends: ResourceRecord
+  abstract: true
+  fields:
+  - name: children
+    doc: ""
+    type: FolderRecord[]?
+  - name: acl
+    doc: ""
+    type: AccessControlListRecord[]?
+
+- name: ProjectRecord
+  doc: |
+    The type for a Synapse project.
+  type: record
+  extends: EntityRecord
+  documentRoot: true
+  docParent: "#RootDoc"
+  fields:
+  - name: type
+    doc: ""
+    jsonldPredicate: syn:type
```

Comment:
What are these `jsonldPredicate`?

Reply:
I'm still trying to fully wrap my mind around `jsonldPredicate`. As far as I can tell, setting `jsonldPredicate: syn:type` for each instance of the `type` field indicates that they all semantically represent the same "thing". If I omit the JSON-LD predicates, I get the following error:

```
schema_salad.exceptions.SchemaException: Predicate collision on type, 'https://synapse.org/#ProjectRecord/type' != 'https://synapse.org/#FolderRecord/type'
```

Since both URIs get shortened to `type`, it's considered a conflict. In this case, they both represent the same attribute/property. Setting `jsonldPredicate: syn:type` means that both fields are identified instead by the following URI:

```
https://synapse.org/#type
```

Reply:
It would be nice if the `syn:type` could actually link to our rest docs that display all the valid entity types...  Although i'm trying to traverse through our rest docs, but I can't seem to find a list of supported concrete types.

Reply:
I had the same thought and ended with the same conclusion. I [asked](https://sagebionetworks.slack.com/archives/C0AFCTGEQ/p1626803197205600) if such a page exists on Slack. If not, we can create our own list of models/types using your OpenAPI spec.